packages/Meshfree/test/tstDetailsCommunicationHelpers.cpp

```diff
@@ -0,0 +1,172 @@
+
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Tpetra_Distributor.hpp>
+
+template <
+    typename View,
+    typename std::enable_if<
+        std::is_same<
+            typename View::traits::memory_space,
+            typename View::traits::host_mirror_space::memory_space>::value,
+        int>::type = 0>
+Teuchos::ArrayView<typename View::const_value_type> toArray( View const &v )
+{
+    return Teuchos::ArrayView<typename View::const_value_type>( v.data(),
+                                                                v.size() );
+}
+
+template <
+    typename View,
+    typename std::enable_if<
+        !std::is_same<
+            typename View::traits::memory_space,
+            typename View::traits::host_mirror_space::memory_space>::value,
+        int>::type = 0>
+Teuchos::Array<typename View::non_const_value_type> toArray( View const &v )
+{
+    auto v_h = Kokkos::create_mirror_view( v );
+    Kokkos::deep_copy( v_h, v );
+
+    return Teuchos::Array<typename View::non_const_value_type>(
+        Teuchos::ArrayView<typename View::const_value_type>( v_h.data(),
+                                                             v_h.size() ) );
+}
+
+// NOTE Shameless hack to get DeviceType from test driver because I don't want
+// to deduce it
+template <typename DeviceType>
+struct Helper
+{
+    template <typename View1, typename View2>
+    static void
+    checkSendAcrossNetwork( Teuchos::RCP<Teuchos::Comm<int> const> const &comm,
+                            View1 const &ranks, View2 const &v_exp,
+                            View2 const &v_ref, bool &success,
+                            Teuchos::FancyOStream &out )
+    {
+        Tpetra::Distributor distributor( comm );
+        distributor.createFromSends( toArray( ranks ) );
+
+        // NOTE here we assume that the reference solution is sized properly
+        auto v_imp =
+            Kokkos::create_mirror( typename View2::memory_space(), v_ref );
+
+        DataTransferKit::DistributedSearchTreeImpl<
+            DeviceType>::sendAcrossNetwork( distributor, v_exp, v_imp );
+
+        // FIXME not sure why I need that guy but I do get a bus error when it
+        // is not here...
+        Kokkos::fence();
+
+        TEST_COMPARE_ARRAYS( toArray( v_imp ), toArray( v_ref ) );
+    }
+};
+
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DetailsDistributedSearchTreeImpl,
+                                   send_across_network, DeviceType )
+{
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    Teuchos::RCP<const Teuchos::Comm<int>> comm =
+        Teuchos::DefaultComm<int>::getComm();
+    int const comm_rank = comm->getRank();
+    int const comm_size = comm->getSize();
+
+    int const DIM = 3;
+
+    // send 1 packet to rank k
+    // receive comm_size packets
+    Kokkos::View<int **, DeviceType> u_exp( "u_exp", comm_size, DIM );
+    Kokkos::parallel_for( "",
```

Comment:
I don't think that you have to give a name if you don't want to. BTW I am fine with not giving any name in test.

Reply:
I forgot about the version of `parallel_for()` that does not take a label.  I will change it.